src/controllers/userController.ts

```diff
@@ -2,36 +2,247 @@ import { Request, Response } from 'express';
+export const deleteUser = async (req: Request, res: Response) => {
+  try {
+    const deleteUserAccount = await User.destroy({
+      where: {
+        id: req.params.id,
+      },
+    });
+    if (deleteUserAccount === 0) {
+      return res.status(404).json({ ok: false, message: 'User with this ID does not exist' });
+    }
+    return res.status(200).json({ ok: true, message: 'User deleted successfully' });
+  } catch (error) {
+    logger.error('Error deleting user: ', error);
+    sendInternalErrorResponse(res, error);
+    return;
+  }
+};
+// Function to edit user role
+export const editUserRole = async (req: Request, res: Response) => {
+  try {
+    const { userId } = req.params;
+    const { roleId } = req.body;
+    if (validateFields(req, ['roleId']).length !== 0) {
+      res.status(400).json({ ok: false, error: 'roleId is required' });
+      return;
+    }
+    const user = await User.findByPk(userId);
+
+    if (!user) {
+      logger.error('Error User Not Found');
+      res.status(404).json({ ok: false, error: 'User not found' });
+    }
+
+    await user?.update({ RoleId: roleId });
+
+    res.status(200).json({ ok: true, message: 'Role assigned successfully.' });
+  } catch (error) {
+    logger.error('Error Edit User Role: ', error);
+    sendInternalErrorResponse(res, error);
   }
 };
+// Function to update user profile
+export const editUser = async (req: Request, res: Response) => {
+  try {
+    const { firstName, lastName, gender, phoneNumber } = req.body;
+
+    const requiredFields = ['firstName', 'lastName', 'gender', 'phoneNumber'];
+    const missingFields = validateFields(req, requiredFields);
+
+    if (missingFields.length > 0) {
+      logger.error(`Required fields are missing:${missingFields.join(', ')}`);
+      res.status(400).json({
+        ok: false,
+        message: `Required fields are missing: ${missingFields.join(', ')}`,
+      });
+      return;
+    }
+
+    if (req.body.email || req.body.password) {
+      res.status(403).json({ ok: false, message: 'It does not possible to update email and password here' });
+    }
+    // when autherization applyed i will be able to access req.user
+    // const user: User | null = await User.findByPk(req.user.dataValues.id) // uncomment when auth middleware
+    const user = await User.findOne({ where: { id: req.params.id } }); // delete this after auth middleware a
 
-export { createUser };
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let uploadedImage: any;
+    if (!req.file) {
+      res.status(400).json({ ok: false, message: 'Profile Image required.' });
+    }
+    if (req.file) {
+      uploadedImage = await uploadImage(req.file.buffer);
+    }
+    const updatedUser = {
+      firstName: firstName || user?.firstName,
+      lastName: lastName || user?.lastName,
+      gender: gender || user?.gender,
+      phoneNumber: phoneNumber || user?.phoneNumber,
+      photoUrl: uploadedImage || user?.photoUrl,
+    };
```

Comment:
Well done. Therefore, remove the upper validations as I mentioned, because even if the fields are not provided in the body, they cannot be empty; they will still contain the data from the database.